src/droplet.F

```diff
@@ -1540,6 +1574,124 @@ subroutine interpolate_to_parcel(np,nrkp,pdata_locind,iflag,jflag,kflag,x3d,y3d,
 
       end subroutine interpolate_to_parcel
 
+      subroutine droplet_diag(rtime,nrec,zh,zf,pdata,pdata_locind)
+
+      use input
+      !use input, only : nparcels,npvals,prx,pry,prz,prsig,pract,prvpx,prvpy,prvpz,prtp,prrp,prmult,prms,przs,pru,prv,prw,prt,prqv,prprs,prrho
+#ifdef NETCDF
+      use writeout_nc_module, only : writedropdiag_nc
+#endif
+      implicit none
+
+      integer, intent(inout) :: nrec
+      real, intent(in) :: rtime
+      real, intent(in), dimension(nparcels,npvals) :: pdata
+      integer, intent(inout), dimension(nparcels,3) :: pdata_locind    ! x/y/z location index of each parcel
+      real, intent(in), dimension(ib:ie,jb:je,kb:ke) :: zh
+      real, intent(in), dimension(ib:ie,jb:je,kb:ke+1) :: zf
+
+      integer :: np,iz
+      integer :: kflag
+      real :: dumzh(kb:ke),dumzf(kb:ke+1),dzf,xl,yl
+
+      integer :: tnumpart,tnumdrop,tnumaerosol
+      real :: radavg,radsqr,Tpavg,Tpsqr,Tfavg,qfavg,prsavg,rhoavg
+      real :: partnum(nk),numconc(nk)
+
+
+
+      !Compute the various types of droplet statistics
+
+
+      !!!!!!! 0th order  !!!!!!!!
+
+      tnumpart = 0
+      radavg = 0.0
+      radsqr = 0.0
+      Tpavg = 0.0
+      Tpsqr = 0.0
+      Tfavg = 0.0
+      qfavg = 0.0
+      prsavg = 0.0
+      rhoavg = 0.0
+
+      do np=1,nparcels
+         if (pdata(np,pract) .gt. 0.0) then
+            
+            tnumpart = tnumpart + 1
+            radavg = radavg + pdata(np,prrp)
+            radsqr = radsqr + pdata(np,prrp)**2
+            Tpavg = Tpavg + pdata(np,prtp)
+            Tpsqr = Tpsqr + pdata(np,prtp)**2 
+            Tfavg = Tfavg + pdata(np,prt)
+            qfavg = qfavg + pdata(np,prqv)
+            prsavg = prsavg + pdata(np,prprs)
+            rhoavg = rhoavg + pdata(np,prrho)
+
+
+         end if
+      end do
+
+      radavg = radavg/tnumpart
+      radsqr = radsqr/tnumpart
+      Tpavg = Tpavg/tnumpart
+      Tpsqr = Tpsqr/tnumpart
+      Tfavg = Tfavg/tnumpart
+      qfavg = qfavg/tnumpart
+      prsavg = prsavg/tnumpart
+      rhoavg = rhoavg/tnumpart
+      
+
+      !!!!!!! 1st order  !!!!!!!!
+      do iz=kb,ke
+         dumzh(iz) = zh(1,1,iz)
+      enddo
+      do iz=kb,ke+1
+         dumzf(iz) = zf(1,1,iz)
+      enddo
+
+      partnum = 0.0
+      numconc = 0.0
+      do np=1,nparcels
+         if (pdata(np,pract) .gt. 0.0) then
+            
+            kflag = 1
+            call find_vertical_location_index (pdata_locind(np,3),pdata(np,prz),kb,ke+1,dumzf(:),kflag,.TRUE.)
+            partnum(pdata_locind(np,3)) = partnum(pdata_locind(np,3)) + 1.0
+
+         end if
+      end do
+
+     if (stretch_x .eq. 0) then
+        xl = dx*nx
+     else
+        xl = tot_x_len
+     endif
+     if (stretch_y .eq. 0) then
+        yl = dy*nx
+     else
+        yl = tot_y_len
+     endif
+
+     do iz=1,nk
+        dzf = dumzf(iz+1)-dumzf(iz)
+        numconc(iz) = partnum(iz)/dzf/xl/yl
+        write(*,'(a15,2e15.6)') 'DROPLET_DIAG1:',dumzh(iz),numconc(iz)
```

Comment:
I would suggest using the format of `DROPLET_DIAG::variable value` here to make the diagnostic output clear in the output log.